text/0083-component.md

```diff
@@ -0,0 +1,115 @@
+
+The main motivation for this is to better model telemetry coming from different
+instrumentation libraries inside an Application (`Resource`). This change
+affects only the OpenTelemetry protocol and exporters, and it does not require
+any API changes.
+
+The proposal is to make `InstrumentationLibrary` a first class concept in
+OpenTelemetry, with a scope defined by the `named Tracer|Meter`. It describes
+all telemetry generated by one `named Tracer|Meter`.
+
+## Explanation
+
+```
+                                Application
++--------------------------------------------------------------------------+
+|                         TracerProvider(Resource)                         |
+|                         MeterProvider(Resource)                          |
+|                                                                          |
+|      Instrumentation Library 1           Instrumentation Library 2       |
+|  +--------------------------------+  +--------------------------------+  |
+|  | Tracer(InstrumentationLibrary) |  | Tracer(InstrumentationLibrary) |  |
+|  | Meter(InstrumentationLibrary)  |  | Meter(InstrumentationLibrary)  |  |
+|  +--------------------------------+  +--------------------------------+  |
+|                                                                          |
+|      Instrumentation Library 3           Instrumentation Library 4       |
+|  +--------------------------------+  +--------------------------------+  |
+|  | Tracer(InstrumentationLibrary) |  | Tracer(InstrumentationLibrary) |  |
+|  | Meter(InstrumentationLibrary)  |  | Meter(InstrumentationLibrary)  |  |
+|  +--------------------------------+  +--------------------------------+  |
+|                                                                          |
++--------------------------------------------------------------------------+
+```
+
+Every application has one `TracerProvider` and one `MeterProvider` which have a
+`Resource` associated with them that describes the Application.
+
+Every instrumentation library has one `Tracer` and one `Meter` which have a
+`InstrumentationLibrary` associated with them that describes the instrumentation
+library.
+
+In case of multi-application deployments like Spring boot (or old Java
+Application servers) every Application will have it's own `TracerProvider` and
+`MeterProvider` instances.
+
+## Internal details
+
+This change affects only the OpenTelemtry protocol, and propose a way to
+represent the telemetry data in a structural way, here is an example for
+metrics:
+
+```proto
+// Resource information.
+message Resource {
+  // Set of labels that describe the resource.
+  repeated opentelemetry.proto.common.v1.AttributeKeyValue attributes = 1;
+}
+
+// InstrumentationLibrary is a message representing the instrumentation library
+// informations such as the fully qualified name and version.
+message InstrumentationLibrary {
+  string name = 1;
+  string version = 2;
+  // Can be extended with attributes.
```

Comment:
Would you call these "attributes" properties of the instrumentation library? Or could they be arbitrary resources?